### DISCUSSION TITLE
Would it make sense to add onFocus and onBlur props to TVFocusGuideView?

### DISCUSSION THREAD

**Discussion by matias-sandell** [react-native-tvos/react-native-tvos#491] [Q&A] [answered]
One use case would be to be able to collapse a left menu when the focus leaves a TVFocusGuideView wrapping the menu items.

Other ways to do this could be to collapse the menu when an item to the right of the menu gets focus or listening for the `right` eventType in a useTVEventHandler() handler. Maybe even debouncing an onBlur function on the menu items themselves. But none of these feel as clean to me as if we had onFocus/onBlur on the TVFocusGuideView itself.

CC @yusufyildirim who has done some great focus PR:s in this repo.

**Comment by douglowder** [comment, marked as answer]
At least on tvOS, the focus guide never actually gets focus itself, so I don't believe the native lifecycle method (`didUpdateFocusInContext:withAnimationCoordinator:`) gets called on it.

**Reply by matias-sandell** [reply]
I see, thanks!

**Comment by yusufyildirim** [comment]
Yeah, that's true. Still, there's a way to detect if one of its children has the focus or not. It'd probably be useful as well. But the use cases are limited I think, it's not super necessary.

I'm working on a separate library that'll be built on top of the core focus stuff and I've been meaning to explore this. It should be relatively easy problem to solve on the JS land instead of adding any extra complexity on the native layer

**Reply by matias-sandell** [reply]
Exciting, looking forward to that library!
